### PR TITLE
Fixes for reproducibility, failed builds, and CI. More hrrr tests, shorter rap&hrrr forecast lengths.

### LINKED ISSUE #649: Some different MPI layouts in regional does not reproduce results
## Description
When used to test different MPI decomposition (like layout 3x2 and 2x3) results are bit identical. Now with layout 10x6, we are getting different results if we use 6x10 or any other layout.

### To Reproduce:
This is in all machines/compilers.

## Output

```
paste the code here (if a short section of log)
```

### LINKED ISSUE #953: Complete feature tests for RAP physics
## Description

PR https://github.com/ufs-community/ufs-weather-model/pull/937 adds several regression tests based on the RAP physics, including feature tests (restart, decomp, 2threads, ...). The following test fails with the RAP suite + GFS SFCDIFF:
```
rap_sfcdiff_decomp
```
The following tests fail with the RAP suite (default MYNN SFC):
```
rap_restart
rap_decomp
```
It seems like that the decomp issue is unrelated to the surface layer scheme, therefore fixing it for one of the tests will likely fix it for the other, too. Once the restart issue for the RAP (MYNN SFC) test is fixed, all RAP SFCDIFF feature tests can be removed (except for `rap_sfcdiff_debug`).

Also, when #937 was merged, the 2thread tests `rap_2threads` were turned off on `jet.intel` and `wcoss_cray` (because of wrong task geometry settings I think). They ran fine on all other platforms. It should be checked if they can be turned on with the correct settings (tasks per node etc.).

### LINKED ISSUE #1249: CI Fix
## Description
Fix broken CI:
1) Update the docker image library to fms 2022.01
2) Add a coupled model test to rt_gnu.conf that excludes the wave component. This is necessary because, with waves, there are not enough cores available on aws using a single node.

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR  are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.  **A new test, hrrr_control_debug, needs a baseline. Many hrrr and rap tests have shorter forecast lengths, and will need baselines as well. Any test using the GF scheme will need a new baseline. Other PRs merged into this one add new tests which will need baselines.**

- [ ] ~~New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.~~

## Description

This PR combines several bugfix PRs into one.

#### Original changes:

1. Fixes a bug in the GF scheme that broke the rap_decomp regression test.
2. Re-enables the rap_decomp regression test.
3. Adds decomp, 2threads, restart, and debug variants of hrrr_control.
4. Reduces the forecast length of rap and hrrr tests from 24 hours to 12, to compensate for new tests. Restarts happen at hour 6 instead of 12.

#### Merged from https://github.com/ufs-community/ufs-weather-model/pull/1263:

Some fixes are made to dycore to resolve the issue with the regional decomposition reproducibility

#### Merged from https://github.com/ufs-community/ufs-weather-model/pull/1253:
Fixes broken CI: 1) Update fms library in docker image; 2) Add cpld noaero nowave test back to rt_gnu.conf. This is to reduce the number of cores used by CI on 1-node AWS EC2 instance.

New baselines are needed for the 3 new tests added to rt.conf:
cpld_control_noaero_p8
cpld_control_nowave_noaero_p8
cpld_debug_noaero_p8

New baselines are needed for the 1 new test added to rt_gnu.conf:
cpld_control_nowave_noaero_p8

#### Merged from https://github.com/ufs-community/ufs-weather-model/pull/1254:

A recent addition of "module" statements at the top of some files broke moninshoc. This adds a dependency so moninshoc will compile. There are three changes:

1. moninshoc.meta has the necessary dependency
2. rt.conf will compile the FV3_GFS_cpld_rasmgshocnsstnoahmp_ugwp suite to ensure moninshoc builds
3. rt_gnu.conf will compile the FV3_GFS_cpld_rasmgshocnsstnoahmp_ugwp suite to ensure moninshoc builds

### Issue(s) addressed

Fixes https://github.com/ufs-community/ufs-weather-model/issues/1249 due to merged PR https://github.com/ufs-community/ufs-weather-model/pull/1253

Fixes https://github.com/NCAR/ccpp-physics/issues/935 due to merged PR https://github.com/ufs-community/ufs-weather-model/pull/1254

Fixes https://github.com/ufs-community/ufs-weather-model/issues/649 due to merged PR https://github.com/ufs-community/ufs-weather-model/pull/1263

Fixes https://github.com/ufs-community/ufs-weather-model/issues/953

This does not fix, but does relate to, #1222, which points out several other FV3_HRRR suite tests do not pass in restart, decomp, 2threads, or debug mode. Apparently the original hrrr_control does pass those tests.

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss2.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI: https://github.com/ufs-community/ufs-weather-model/actions/runs/2511287577

## Dependencies

https://github.com/NCAR/ccpp-physics/pull/942
https://github.com/NOAA-EMC/fv3atm/pull/553
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/194

### CODE REVIEW

**Comment by SamuelTrahanNOAA** [comment]
Ideally, this should be combined with https://github.com/ufs-community/ufs-weather-model/issues/953 which enables all rap_control tests.

**Comment by climbfuji** [comment]
> Ideally, this should be combined with #953 which enables all rap_control tests.

@SamuelTrahanNOAA Feel free to pull my changes for ufs-weather-model into your PR, or we can do it the other way round.

**Comment by SamuelTrahanNOAA** [comment]
> > Ideally, this should be combined with #953 which enables all rap_control tests.
> 
> @SamuelTrahanNOAA Feel free to pull my changes for ufs-weather-model into your PR, or we can do it the other way round.

I will merge it to my branch shortly.

**Comment by SamuelTrahanNOAA** [comment]
@climbfuji Your FV3 is out of date. Do you want me to copy your submodule pull requests too?

**Comment by climbfuji** [comment]
I can also update my branches, how urgent is it?

**Comment by SamuelTrahanNOAA** [comment]
It won't be urgent until the day this is scheduled for commit. If you won't be available on demand that day, it would be best if I submit replacements for your PRs.

**Comment by climbfuji** [comment]
If you want to create replacement PRs and pull in my changes, please go ahead - makes my life easier, but creates more work for you. Thanks in advance if you decide to do that :-)

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA We may combine a few PRs to here: #1254 and #1253. @MinsukJi-NOAA Can you make PR to Sam's branch?

**Comment by SamuelTrahanNOAA** [comment]
I have one last change to make to this branch before final testing. There are two tests disabled on Jet that should not be. I'm going to test them now to make sure they work, and then I'll push the new rt.conf and rt_gnu.conf.

**Comment by MinsukJi-NOAA** [comment]
> @SamuelTrahanNOAA We may combine a few PRs to here: #1254 and #1253. @MinsukJi-NOAA Can you make PR to Sam's branch?

I will make a PR once @SamuelTrahanNOAA 's branch is up to date with the UWM develop branch.

**Comment by SamuelTrahanNOAA** [comment]
The branch is up to date.

**Comment by junwang-noaa** [comment]
@SamuelTrahanNOAA Can we combine #1263 with your PR? #1263 has some decomposition fixed for regional parallel tests. Thanks.

**Comment by SamuelTrahanNOAA** [comment]
> @SamuelTrahanNOAA Can we combine #1263 with your PR? #1263 has some decomposition fixed for regional parallel tests. Thanks.

It is in my mental queue. I am testing #1254 now, and I'll do #1253 next, and then #1263.

**Comment by SamuelTrahanNOAA** [comment]
@junwang-noaa I think they're all merged into this branch, but it would be best if others check my work before final testing.

**Comment by junwang-noaa** [comment]
@SamuelTrahanNOAA The code changes from PR#1263 and #1254 look good to me. Thanks

**Comment by SamuelTrahanNOAA** [comment]
I'm testing this menagerie on jet.intel and hera.gnu now.

**Comment by MinsukJi-NOAA** [comment]
> @junwang-noaa I think they're all merged into this branch, but it would be best if others check my work before final testing.

@SamuelTrahanNOAA I do not see changes from #1253

File: tests/tests/hrrr_control_2threads [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                   atmf000.nc \
+                   atmf009.nc \
+                   atmf012.nc \
+                   GFSFLX.GrbF00 \
+                   GFSFLX.GrbF09 \
+                   GFSFLX.GrbF12 \
+                   GFSPRS.GrbF00 \
+                   GFSPRS.GrbF09 \
+                   GFSPRS.GrbF12 \
+                   RESTART/coupler.res \
```
Comment (DusanJovic-NOAA) [inline tests/tests/hrrr_control_2threads:23]: Unless this test is going to be used to run subsequent restart test, which I think is not the case here, there's no need to save restart files in a baseline.

File: tests/tests/hrrr_control_decomp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                   atmf000.nc \
+                   atmf009.nc \
+                   atmf012.nc \
+                   GFSFLX.GrbF00 \
+                   GFSFLX.GrbF09 \
+                   GFSFLX.GrbF12 \
+                   GFSPRS.GrbF00 \
+                   GFSPRS.GrbF09 \
+                   GFSPRS.GrbF12 \
+                   RESTART/coupler.res \
```
Comment (DusanJovic-NOAA) [inline tests/tests/hrrr_control_decomp:23]: Same as above, I do not think we need to save restart files in a baseline.

File: tests/tests/hrrr_control_2threads (hunk 1 above)
Comment (SamuelTrahanNOAA) [reply tests/tests/hrrr_control_2threads:23]: This test does not have its own baseline. It uses hrrr_control. If you look at other decomp and 2threads variants, such as cpld_2threads_p8, you will see they do compare the restart files against their control.

```
export CNTL_DIR=hrrr_control
```

File: tests/tests/hrrr_control_decomp (hunk 2 above)
Comment (SamuelTrahanNOAA) [reply tests/tests/hrrr_control_decomp:23]: This test does not have its own baseline. It uses hrrr_control. If you look at other decomp and 2threads variants, such as cpld_2threads_p8, you will see they do compare the restart files against their control.

```
export CNTL_DIR=hrrr_control
```

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA Sorry, it should be merged now.

File: tests/tests/hrrr_control_decomp (hunk 2 above)
Comment (DusanJovic-NOAA) [reply tests/tests/hrrr_control_decomp:23]: What I meant to say is that we do not need to have them listed here and compare them against the baselines, if history files are identical. Comparing all these restart files takes time.

Comment (SamuelTrahanNOAA) [reply tests/tests/hrrr_control_decomp:23]: The rap variants (rap_2threads, etc.) already do those extra comparisons. I want the hrrr tests to match the rap tests.

Comment (DusanJovic-NOAA) [reply tests/tests/hrrr_control_decomp:23]: Ok. Then leave it. But, I suggest at some point we save (list) restart files only in _control and _restart variants, and remove from others.

Comment (SamuelTrahanNOAA) [reply tests/tests/hrrr_control_decomp:23]: You should open an issue for that.

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA As the PR is already tested ok on hera/jet, I think we can start creating new baseline. Let me know if you still want last min check.

**Comment by SamuelTrahanNOAA** [comment]
It works, as far as I can tell. I generated baselines and ran regression tests on hera.intel, hera.gnu, and jet.intel, successfully.

**Comment by MinsukJi-NOAA** [comment]
@SamuelTrahanNOAA Could you please make this one line change? This change is to use noaaemc docker hub account instead of my personal account for input data, and only affects CI.
```
diff --git a/tests/ci/ci.sh b/tests/ci/ci.sh
index 34abe8ec..c93a865c 100755
--- a/tests/ci/ci.sh
+++ b/tests/ci/ci.sh
@@ -75,7 +75,7 @@ if [ $BUILD = "true" ]; then
 elif [ $RUN == "true" ]; then
 
   docker volume rm -f DataVolume >/dev/null &&
-    docker run -d --rm -v DataVolume:/tmp minsukjinoaa/input-data:20220414
+    docker run -d --rm -v DataVolume:/tmp noaaemc/input-data:20220414
 
   docker create -u builder -e "CI_TEST=true" -e "USER=builder" \
                 -e "RT_MACHINE=linux.gnu" -e "RT_COMPILER=gnu" \
```

**Comment by BrianCurtis-NOAA** [comment]
I have an error on WCOSS2: /lfs/h2/emc/ptmp/brian.curtis/FV3_RT/rt_231410/cpld_debug_noaero_p8
```
+ mpiexec -n 200 -ppn 128 -depth 1 ./fv3.exe
forrtl: error (182): floating invalid - possible uninitialized real/complex variable.
Image              PC                Routine            Line        Source             
fv3.exe            000000000E11F19B  Unknown               Unknown  Unknown
libpthread-2.31.s  000015140CD648C0  Unknown               Unknown  Unknown
fv3.exe            000000000BB32F2B  ice_comp_nuopc_mp         186  ice_comp_nuopc.F90
fv3.exe            0000000000DA9AD6  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
fv3.exe            0000000000DAD74A  ESMCI_FTableCallE         824  ESMCI_FTable.C
fv3.exe            000000000138A68F  _ZN5ESMCI3VMK5ent        2308  ESMCI_VMKernel.C
fv3.exe            0000000001374039  _ZN5ESMCI2VM5ente        1216  ESMCI_VM.C
fv3.exe            0000000000DAAF57  c_esmc_ftablecall         981  ESMCI_FTable.C
fv3.exe            00000000009A6A7D  esmf_compmod_mp_e        1222  ESMF_Comp.F90
fv3.exe            00000000007CCA76  esmf_gridcompmod_        1407  ESMF_GridComp.F90
fv3.exe            0000000000462D5F  nuopc_driver_mp_l        2565  NUOPC_Driver.F90
fv3.exe            0000000000487D32  nuopc_driver_mp_i        1264  NUOPC_Driver.F90
fv3.exe            0000000000490D35  nuopc_driver_mp_i         455  NUOPC_Driver.F90
fv3.exe            0000000000DA9AD6  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
fv3.exe            0000000000DAD74A  ESMCI_FTableCallE         824  ESMCI_FTable.C
fv3.exe            000000000138A68F  _ZN5ESMCI3VMK5ent        2308  ESMCI_VMKernel.C
fv3.exe            0000000001374039  _ZN5ESMCI2VM5ente        1216  ESMCI_VM.C
fv3.exe            0000000000DAAF57  c_esmc_ftablecall         981  ESMCI_FTable.C
fv3.exe            00000000009A6A7D  esmf_compmod_mp_e        1222  ESMF_Comp.F90
fv3.exe            00000000007CCA76  esmf_gridcompmod_        1407  ESMF_GridComp.F90
fv3.exe            000000000042A404  MAIN__                    381  UFS.F90
fv3.exe            00000000004271D2  Unknown               Unknown  Unknown
libc-2.31.so       000015140C7752BD  __libc_start_main     Unknowforrtl: error (182): floating invalid - 
possible uninitialized real/complex variable.
```

**Comment by BrianCurtis-NOAA** [comment]
@SamuelTrahanNOAA if you have access to WCOSS2: /lfs/h2/emc/ptmp/brian.curtis/FV3_RT/rt_231410/cpld_debug_noaero_p8

**Comment by SamuelTrahanNOAA** [comment]
@BrianCurtis-NOAA I don't see how Dom's bugfix could cause that. Maybe @junwang-noaa should look at it and tell whether the atmos_cubed_sphere bugfix is related?

Presently, I cannot access WCOSS2, and it would probably take a week and a half to fix that since my supervisor is on vacation.

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA I don't see how Dom's bugfix could cause that. Maybe @junwang-noaa should look at it and tell whether the atmos_cubed_sphere bugfix is related?
> 
> Presently, I cannot access WCOSS2, and it would probably take a week and a half to fix that since my supervisor is on vacation.

OK, I will look into it more. Thanks.

**Comment by SamuelTrahanNOAA** [comment]
That configuration doesn't use the GF scheme, nor moninshoc. Assuming these bug fixes are the cause, that leaves only CI and the regional decomposition fix as suspects.

**Comment by BrianCurtis-NOAA** [comment]
> That configuration doesn't use the GF scheme, nor moninshoc. Assuming these bug fixes are the cause, that leaves only CI and the regional decomposition fix as suspects.

OK. I'm also starting fresh considering the possibility of a stale clone in my directory. I'll let you know.

**Comment by MinsukJi-NOAA** [comment]
> > That configuration doesn't use the GF scheme, nor moninshoc. Assuming these bug fixes are the cause, that leaves only CI and the regional decomposition fix as suspects.
> 
> OK. I'm also starting fresh considering the possibility of a stale clone in my directory. I'll let you know.

@BrianCurtis-NOAA  @SamuelTrahanNOAA `cpld_debug_noaero_p8` is a new test added to `rt.conf`. This test was previously only on `rt_gnu.conf`. I was able to successfully run this test on hera.intel.

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA I have pushed your change to my branch. For future note, you *should* be able to push to my PR branches yourself. I don't know if you've tested that yet though.

**Comment by junwang-noaa** [comment]
@SamuelTrahanNOAA @BrianCurtis-NOAA The dycore changes are for regional case, it does not have impact on global tests

**Comment by junwang-noaa** [comment]
@MinsukJi-NOAA We are still using Intel 2018 on wcoss2, could that be the reason?

**Comment by MinsukJi-NOAA** [comment]
From #1253:

New baselines are needed for the 3 new tests added to rt.conf:
cpld_control_noaero_p8
cpld_control_nowave_noaero_p8
cpld_debug_noaero_p8

New baselines are needed for the 1 new test added to rt_gnu.conf:
cpld_control_nowave_noaero_p8

**Comment by SamuelTrahanNOAA** [comment]
@MinsukJi-NOAA Your PR ("CI fix") added the test that is failing. You added these tests: cpld_debug_noaero_p8, cpld_control_noaero_p8,  and cpld_control_nowave_noaero_p8. Could you please check that I added them to my branch correctly?

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA Your PR ("CI fix") added the test that is failing. You added these tests: cpld_debug_noaero_p8, cpld_control_noaero_p8, and cpld_control_nowave_noaero_p8. Could you please check that I added them to my branch correctly?

It looks they were added correctly. Since the test itself is not new, all related files except rt.conf are not new.

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA We are still using Intel 2018 on wcoss2, could that be the reason?

Since wcoss is also using intel 18, let me try to run the test there.

**Comment by MinsukJi-NOAA** [comment]
@SamuelTrahanNOAA @BrianCurtis-NOAA @junwang-noaa `cpld_control_noaero_p8` ran successfully on wcoss dell with Intel 18.
I noticed the following line in rt.conf:
`RUN | cpld_debug_p8 | - wcoss_cray  wcoss2.intel`
Without knowing why `cpld_debug_p8` is not working on cray and wcoss2, we can do a similar thing with `cpld_debug_noaero_p8` for now.
```
-- a/tests/rt.conf
+++ b/tests/rt.conf
@@ -23,8 +23,8 @@ RUN     | cpld_control_nowave_noaero_p8
 COMPILE | -DAPP=S2SWA -DDEBUG=ON -DCCPP_SUITES=FV3_GFS_v17_coupled_p8,FV3_GFS_cpld_rasmgshocnsstnoahmp_ugwp                       | - wcoss_cray  wcoss2.intel              | fv3 |
 RUN     | cpld_debug_p8                                                                                                           | - wcoss_cray  wcoss2.intel              | fv3 |
 
-COMPILE | -DAPP=S2SW -DDEBUG=ON -DCCPP_SUITES=FV3_GFS_v17_coupled_p8                                                              |                                         | fv3 |
-RUN     | cpld_debug_noaero_p8                                                                                                    |                                         | fv3 |
+COMPILE | -DAPP=S2SW -DDEBUG=ON -DCCPP_SUITES=FV3_GFS_v17_coupled_p8                                                              | - wcoss_cray  wcoss2.intel              | fv3 |
+RUN     | cpld_debug_noaero_p8                                                                                                    | - wcoss_cray  wcoss2.intel              | fv3 |
 
 # Waves and aerosol off for computing fluxes in mediator
 COMPILE | -DAPP=S2S -DCCPP_SUITES=FV3_GFS_v17_coupled_p8_sfcocn -DCMEPS_AOFLUX=ON                                                 | - wcoss_cray                            | fv3 |
```

**Comment by SamuelTrahanNOAA** [comment]
I have disabled cpld_debug_noaero_p8 on wcoss.cray and wcoss2.intel

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
[BL] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/959879766/20220616111511/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/959879766/20220616122232/ufs-weather-model
Please make changes and add the following label back: orion-intel-BL

**Comment by SamuelTrahanNOAA** [comment]
Orion is having filesystem issues right now, according to an email from Joey B. Jones three minutes ago.

> This afternoon we have started to see problems with the /work filesystem on Orion. We are investigating the problem and will send a notification when the issue has been resolved.
> 
> _______________________________________
> Joey B. Jones
> HPC2 Systems And Network Administration
> Mississippi State University

**Comment by jkbk2004** [comment]
> Orion is having filesystem issues right now, according to an email from Joey B. Jones three minutes ago.
> 
> > This afternoon we have started to see problems with the /work filesystem on Orion. We are investigating the problem and will send a notification when the issue has been resolved.
> > 
> > Joey B. Jones
> > HPC2 Systems And Network Administration
> > Mississippi State University

@SamuelTrahanNOAA Sure! we can check Orion later! I am moving to other machines.

**Comment by SamuelTrahanNOAA** [comment]
Orion should work now, according to the aforementioned Joey.

> The issues with /work have been corrected. Please contact your respective helpdesk if you see further problems.

**Comment by MinsukJi-NOAA** [comment]
@SamuelTrahanNOAA `hrrr_control_2threads` failed twice on wcoss cray with the same error message:
```
+ aprun -j 1 -n 78 -N 12 -d 2 -cc depth ./fv3.exe
apsched: claim exceeds reservation's node-count
```
Is the run configuration (e.g. number of tasks, and etc.) correct?

**Comment by SamuelTrahanNOAA** [comment]
> @SamuelTrahanNOAA `hrrr_control_2threads` failed twice on wcoss cray with the same error message:
> 
> ```
> + aprun -j 1 -n 78 -N 12 -d 2 -cc depth ./fv3.exe
> apsched: claim exceeds reservation's node-count
> ```
> 
> Is the run configuration (e.g. number of tasks, and etc.) correct?

All of the other 2threads tests are disabled on wcoss.cray, so I just updated rt.conf to disabled hrrr_control_2threads on wcoss.cray as well.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: BL
[BL] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/959879766/20220616204517/ufs-weather-model
[BL] Error: Test control_p8_rrtmgp 022 failed in run_test failed 
Please make changes and add the following label back: jet-intel-BL

**Comment by SamuelTrahanNOAA** [comment]
All of the Jet tests worked for me. There are frequent system issues on Jet, so you may have to submit a job or two in every workflow execution.

**Comment by jkbk2004** [comment]
> All of the Jet tests worked for me. There are frequent system issues on Jet, so you may have to submit a job or two in every workflow execution.

sure!

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA All tests pass and we are good to start merging in dependencies: ccpp -> atmos_cubed_sphere -> fv3atm. Can you start with ccpp first?

**Comment by SamuelTrahanNOAA** [comment]
I have merged ccpp, but I lack the ability to merge atmos_cubed_sphere.

**Comment by jkbk2004** [comment]
> I have merged ccpp, but I lack the ability to merge atmos_cubed_sphere.

I left a note to atmos_cubed_sphere.

**Comment by SamuelTrahanNOAA** [comment]
The fv3atm PR is ready for merge. A few minutes ago, someone merged atmos_cubed_sphere. I've updated the fv3atm branch to point to the top of atmos_cubed_sphere and ccpp-physics.

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA Everything ready upto fv3atm. We need hash update and revert pointer.

**Comment by SamuelTrahanNOAA** [comment]
Done.

**Comment by jkbk2004** [comment]
@DusanJovic-NOAA The PR is ready for final review to approve.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by MinsukJi-NOAA — APPROVED** [review]

**Pull request merged**